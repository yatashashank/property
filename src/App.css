.header, .menu, .content {
  box-sizing: border-box;
}
.content{
}
.card{
  display: grid;
  grid-template-columns: auto auto auto auto;
}

.app {
  height: 100%;
  flex-direction: column;
  display: flex;
}

html, #root, body {
  height: 100%;
  margin: 0px;
}

body {
  font-family: Helvetica, Arial;
}

.app-body {
  display: flex;
  flex:1;
}

.content {
  position: relative;
  flex: 1;
  padding: 10px;
  overflow: auto;
}

.online-shop-title {
  font-size: 26px;
  margin: 5px;
  color: gray;
}

.online-shop-title-smaller {
  font-size: 22px;
  margin: 5px;
  color: gray;
}