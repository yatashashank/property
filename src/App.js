import React, { Component } from "react";
import "./App.css";
import Header from "./Components/Header/Header";
import Menu from "./Components/Menu/Menu";
import { Switch, Route } from "react-router-dom";
import Details from "./Components/Property/details";
import Main from "./Components/Property/main";
class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      show: true
    };
  }
  handleClick = () => {
    this.setState({ show: !this.state.show });
  };
  render() {
    return (
      <div className="app">
        <Header onclick={this.handleClick} />
        <div className="app-body">
          <Menu show={this.state.show} />
          <div className="content">
            <Switch>
              <Route path="/" exact component={Main} />
              <Route path="/details/:id" component={Details} />
            </Switch>
          </div>
        </div>
      </div>
    );
  }
}

export default App;
