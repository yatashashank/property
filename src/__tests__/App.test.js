import React from 'react';
import toJson from 'enzyme-to-json';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import { shallow,render,mount } from 'enzyme';

import App from '../App'
import {Switch,Route,BrowserRouter} from 'react-router-dom'

describe('<App />', () => {

  it('snapshot testing for App component', () => {
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('checking classname for App component ',()=>{
    const wrap=shallow(<App />);
    const tree=wrap.find('.app')
    expect(tree).toHaveLength(1)
  });

  it('class name for main content', () => { 
    const wrap=shallow(<App />);
    const tree=wrap.find('div.content') 
    expect(tree.hasClass('content')).toBeTruthy()
  })

  it('checking total routes and single routes', () => { 
    const wrap=shallow(<App />);
    const tree=wrap.find(Route)
    const tree1=wrap.find(Route).find({path: '/details/:id'}) 
    expect(tree.length).toBe(2)
    expect(tree1.length).toBe(1)
  })


})