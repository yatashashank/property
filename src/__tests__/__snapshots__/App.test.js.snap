// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> snapshot testing for App component 1`] = `
<div
  className="app"
>
  <Header
    onclick={[Function]}
  />
  <div
    className="app-body"
  >
    <Menu
      show={false}
    />
    <div
      className="content"
    >
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          path="/details/:id"
        />
      </Switch>
    </div>
  </div>
</div>
`;
